### ISSUE TITLE
[Feature] Redirect to url

### ISSUE THREAD

**Issue by stalniy** [pshrmn/curi#234]
## Description

I'm in a process of rewriting legacy blog to webcomponents with i18n support. So, I want to preserve old links. Inside "catch all" route I check whether lang is url and if not then I want to redirect to the same url but with default lang

## Current Behavior

Currently I can redirect only to known route name (inside `respond` method of route):

## Wanted behavior

I want to have possibility to redirect to url.

```js
const routes = [
  {
     path: '(.*)',
     respond() {
       if (/* lang is not in url */) {
          return {
             redirect: { url: `/${defaultLocale}${location.pathname}${location.search}${location.hash}` }
          }
       }
       // show 404
     }
  }
]
```

**Comment by pshrmn** [comment]
Hmm, I would have to think about whether I want that to be part of the official API. When a response has a `redirect` property, we end up generating a `url` string and using that to navigate, so it wouldn't be the worst thing to have `url` be an optional property that we use instead of a requiring the route name/params, etc. I think that I used to have something like that, but then I would have to pass in a way for a user to generate a `url` from a route's `name`/`params`.

In the mean time, there are a couple of possible approaches that you could take.

1. Your catch all route could match multiple segments and you could redirect to that route with an array of path segments as the param. This was my initial thought, and may work, but it is pretty hacky.

https://codesandbox.io/s/curi-in-app-url-redirect-ctsfg

2. Another option would be to have the catch all route return a response with the URL to redirect to attached to the response's data. Then in your root component, you could detect that data and redirect yourself. This is a bit more hands on, but maybe less hacky.

What sorts of URLs do you need to preserve? If they are all blog posts that have a common format, would you not be able to just have a route that matches the old blog post URL format and redirects?

```ts
const routes = prepareRoutes([
  { name: 'Post', path: ':lang/:slug', ... },
  {
    name: 'Old Post',
    path: ':slug',
    respond({ match }) {
      return {
        redirect: { name: 'Post', params: match.params, ... }
      };
    }
  }
 },
]);
```

---

I haven't been giving Curi much love lately, but I'm hoping to find the time to give it a good evaluation to figure out what is working and what could be improved. I'll certainly keep this in mind when I do that.

**Comment by stalniy** [comment]
I can make a PR with optional `url` option if you ok with that. Currently I also do something Jacky there :) https://github.com/stalniy/freaksidea-pwa/blob/master/src/config/routes.js#L79

**Comment by pshrmn** [comment]
I just published `@curi/router` v2.1.0, which allows you to use `redirect: { url: '/some-place' }`. I haven't written documentation yet (and just realized that I didn't update the changelog), but it's there. If you need search/hash, you have to include them in the url string yourself instead of providing them as separate properties.

```ts
respond({ match }) {
  const { location } = match.
  return {
    redirect: {
      url: `${DEFAULT_LANG}/${location.pathname}${location.search}${location.hash}`,
    }
  };
}
```

Please let me know if there are any issues with it, but I believe that this solves your use case.

**Comment by stalniy** [comment]
Awesome! Much appreciated!
